### PR TITLE
ref(js): Remove TeamActions

### PR DESCRIPTION
**Pull request by evanpurkhiser**

### CODE REVIEW

File: static/app/actionCreators/teams.tsx [hunk 1]
```diff
@@ -30,47 +29,30 @@ type OrgAndTeamSlug = OrgSlug & {teamId: string};
  */
 type MemberId = {memberId: string};
 
-// Fetch teams for org
```
Comment (evanpurkhiser) [inline static/app/actionCreators/teams.tsx:33]: Totally unused.

File: static/app/actionCreators/teams.tsx [hunk 2]
```diff
@@ -81,18 +63,14 @@ export function joinTeam(
   const endpoint = `/organizations/${params.orgId}/members/${
     params.memberId ?? 'me'
   }/teams/${params.teamId}/`;
-  const id = uniqueId();
-
-  TeamActions.update(id, params.teamId);
```
Comment (evanpurkhiser) [inline static/app/actionCreators/teams.tsx:86]: unused

File: static/app/actionCreators/teams.tsx [hunk 3]
```diff
@@ -106,34 +84,28 @@ export function leaveTeam(
...
     },
     error: error => {
-      TeamActions.updateError(id, params.teamId, error);
       doCallback(options, 'error', error);
     },
   });
 }
 
 export function createTeam(api: Client, team: Pick<Team, 'slug'>, params: OrgSlug) {
-  TeamActions.createTeam(team);
```
Comment (evanpurkhiser) [inline static/app/actionCreators/teams.tsx:127]: unused

File: static/app/stores/projectsStore.tsx [hunk 4]
```diff
@@ -72,10 +71,6 @@ const storeConfig: ProjectsStoreDefinition = {
     this.unsubscribeListeners.push(
       this.listenTo(ProjectActions.updateSuccess, this.onUpdateSuccess)
     );
-
-    this.unsubscribeListeners.push(
```
Comment (evanpurkhiser) [inline static/app/stores/projectsStore.tsx:76]: Directly called from the `TeamStore.onRemoveSuccess` now

File: static/app/actionCreators/teams.tsx (hunk 2 above)
Comment (priscilawebdev) [reply static/app/actionCreators/teams.tsx:86]: good catch!

File: static/app/actionCreators/teams.tsx (hunk 3 above)
Comment (priscilawebdev) [reply static/app/actionCreators/teams.tsx:127]: here too

**Review by priscilawebdev — APPROVED** [review]
lgtm

**Pull request merged**